File: Hashmap/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int mapa[26]; memset(mapa, 0, sizeof(mapa));

        for(auto &x: s) mapa[x-'a']++;
        for(auto &x: t){
            mapa[x-'a']--;
            if(mapa[x-'a'] < 0) return false;
        }

        for(int i = 0; i<26; i++) if(mapa[i]!=0) return false;
        return true;

    }
};

File: Hashmap/valid-anagram.rs
use std::collections::HashMap;

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        let mut mapa: HashMap<char, u16> = HashMap::new();

        for ch in s.chars() {
            *mapa.entry(ch).or_insert(0) += 1;
        }

        for ch in t.chars() {
            *mapa.entry(ch).or_insert(0) -= 1;
        }

        for &valor in mapa.values() {
            if valor != 0 {
                return false;
            }
        }

        return true;

        
    }
}